toqito/matrices/pauli.py

```diff
@@ -92,9 +91,6 @@ def pauli(ind: int | str | list[int] | list[str], is_sparse: bool = False) -> np
         else:
             pauli_mat = np.identity(2)
 
-        if is_sparse:
-            pauli_mat = sparse.csr_matrix(pauli_mat)  # pylint: disable=redefined-variable-type
```

Comment:
@Bchass I think instead of removing `is_sparse` option  completely due to `sparse.csr_matrix`, we could try to switch to using `sparse.csr_array` instead. 

https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.csr_array.html#scipy.sparse.csr_array

I haven't tried this out yet, FYI! If you run into a lot of issues, then we could simply remove that option in case we can't find other options.

Reply:
Looked into `sparse.csr_array` and it works a lot better for this use case. No need to alter any test cases.